File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.java
import java.util.*;

class Solution {
    public int solution(int n) {
        int answer = 0;
        
        int count = 0;
        int digit = n;
        while (digit > 0) {
            count++;
            digit /= 10;
        }
        
        // 3진법으로 변환
        Stack<Integer> stk = new Stack<>();
        
        while (n != 0) {
            int k = n%3;
            stk.add(k);
            
            n /= 3;
        }
        
        /*
        while (!stk.isEmpty()) {
            System.out.println(stk.peek());
            stk.pop();
        }
        */
        
        int i = 1;
        while (!stk.isEmpty()) {
            int num = stk.peek();
            while (num -- > 0)
                answer += i;
            
            stk.pop();
            i *= 3;
        }
        return answer;
    }
}

File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.js
function solution(n) {
    let answer = 0;
    
    let stack = [];
    while (n > 0) {
        stack.push(n%3);
        n = Math.floor(n/3);
    }
    
    // stack = [0, 0, 2, 1];
    for (let i = 0; i < stack.length; i++)
        console.log(stack[i]);
    
    let power = 1;
    for (let i = stack.length-1; i >= 0; i--) {
        answer += stack[i] * power;
        power *= 3;
    }
    
    return answer;
}